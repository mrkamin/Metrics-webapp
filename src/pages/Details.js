import React from 'react';
import { useLocation } from 'react-router-dom';
import style from '../style/Details.module.css';

const Details = () => {
  const location = useLocation();
  const {
    symbol, companyName, marketCap, sector,
    industry, beta, price,
    lastAnnualDividend, volume,
    exchange, exchangeShortName,
    country, isEtf, isActivelyTrading,
  } = location.state;

  return (
    <div className={style.homeform}>
      <div className={style.homeHederBackground}>
        <div className={style.homeHeder}>
          <div className={style.detailcompanyName}>CompanyName:</div>
          <div className={style.detailcompanyNameTwo}>
            {companyName}
          </div>
          <span className={style.homeform}>
            MarketCap:
            { marketCap}
          </span>
        </div>
      </div>
      <div className={style.DetailsListContainer}>
        <h2 className={style.homeform}>Stock Screener Details</h2>
        <ul className={style.DetailsUL}>
          <li className={style.DetailsLi}>
            <h4 className={style.homeform}>Sector</h4>
            <p className={style.homeform}>{sector}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>Industry</h4>
            <p>{industry}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>Beta</h4>
            <p>{beta}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>Price</h4>
            <p>{price}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>LastAnnualDividend</h4>
            <p>{lastAnnualDividend}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>Exchange</h4>
            <p>{exchange}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>ExchangeShortName</h4>
            <p>{exchangeShortName}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>Country</h4>
            <p>{country}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>IsEtf</h4>
            <p>{isEtf}</p>
          </li>

          <li className={style.DetailsLi}>
            <h4>IsActivelyTrading</h4>
            <p>{isActivelyTrading}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>symbol</h4>
            <p>{symbol}</p>
          </li>
          <li className={style.DetailsLi}>
            <h4>Volume</h4>
            <p>{volume}</p>
          </li>
        </ul>
      </div>
    </div>

  );
};

export default Details;
