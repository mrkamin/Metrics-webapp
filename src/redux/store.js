import { configureStore } from '@reduxjs/toolkit';
import stockReducer from './Stock-Screener/stockScreener';

const store = configureStore({
  reducer: {
    stockMaterials: stockReducer,
  },
});

export default store;
