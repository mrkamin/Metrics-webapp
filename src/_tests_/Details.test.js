import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Details from '../pages/Details';
import renderProvider from '../redux/Stock-Screener/stockScreener-test';

const mockHistory = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistory,
  }),
  useLocation: () => ({
    pathname: '/details',
    state: { companyName: 'Apple Inc.' },
  }),
}));

it('snapshots', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('testing the heading inside Details component', async () => {
  renderProvider(
    <BrowserRouter>
      <Details />
    </BrowserRouter>,
  );

  expect(screen.getByText(/Stock Screener Details/i)).toBeInTheDocument();
  expect(screen.getByText(/Apple Inc./i)).toBeInTheDocument();
});
