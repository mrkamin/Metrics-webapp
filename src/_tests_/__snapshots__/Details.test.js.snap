// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="homeform"
>
  <div
    className="homeHederBackground"
  >
    <div
      className="homeHeder"
    >
      <div
        className="detailcompanyName"
      >
        CompanyName:
      </div>
      <div
        className="detailcompanyNameTwo"
      >
        Apple Inc.
      </div>
      <span
        className="homeform"
      >
        MarketCap:
      </span>
    </div>
  </div>
  <div
    className="DetailsListContainer"
  >
    <h2
      className="homeform"
    >
      Stock Screener Details
    </h2>
    <ul
      className="DetailsUL"
    >
      <li
        className="DetailsLi"
      >
        <h4
          className="homeform"
        >
          Sector
        </h4>
        <p
          className="homeform"
        />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          Industry
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          Beta
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          Price
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          LastAnnualDividend
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          Exchange
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          ExchangeShortName
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          Country
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          IsEtf
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          IsActivelyTrading
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          symbol
        </h4>
        <p />
      </li>
      <li
        className="DetailsLi"
      >
        <h4>
          Volume
        </h4>
        <p />
      </li>
    </ul>
  </div>
</div>
`;
